server/src/test/java/org/apache/iotdb/db/qp/physical/InsertTabletPlanTest.java

```diff
@@ -186,29 +208,50 @@ public void testInsertTabletPlanWithDeviceTemplate()
       compressionTypes.add(CompressionType.SNAPPY);
     }
 
-    CreateTemplatePlan plan =
-        new CreateTemplatePlan(
-            "template1", measurementList, dataTypesList, encodingList, compressionTypes);
-
-    IoTDB.metaManager.createDeviceTemplate(plan);
-    IoTDB.metaManager.setDeviceTemplate(new SetDeviceTemplatePlan("template1", "root.isp.d1"));
+    return new CreateTemplatePlan(
+        "template1", measurementList, dataTypesList, encodingList, compressionTypes);
+  }
 
-    IoTDBDescriptor.getInstance().getConfig().setAutoCreateSchemaEnabled(false);
+  @Test
+  public void testInsertTabletPlanWithDeviceTemplateAndAutoCreateSchema()
+      throws QueryProcessException, MetadataException, InterruptedException,
+          QueryFilterOptimizationException, StorageEngineException, IOException {
+    CreateTemplatePlan plan = getCreateTemplatePlan();
 
+    IoTDB.metaManager.createDeviceTemplate(plan);
+    IoTDB.metaManager.setDeviceTemplate(new SetDeviceTemplatePlan("template1", "root.isp"));
     InsertTabletPlan tabletPlan = getInsertTabletPlan();
 
     PlanExecutor executor = new PlanExecutor();
     executor.insertTablet(tabletPlan);
 
-    System.out.println(Arrays.toString(tabletPlan.getMeasurementMNodes()));
-
     QueryPlan queryPlan = (QueryPlan) processor.parseSQLToPhysicalPlan("select * from root.isp.d1");
     QueryDataSet dataSet = executor.processQuery(queryPlan, EnvironmentUtils.TEST_QUERY_CONTEXT);
     Assert.assertEquals(3, dataSet.getPaths().size());
     while (dataSet.hasNext()) {
       RowRecord record = dataSet.next();
       Assert.assertEquals(6, record.getFields().size());
     }
+
+    // test recover
+    // we want to recover
+    EnvironmentUtils.stopDaemon();
+    // wait for close
```

Comment:
You should use 
```
IoTDB.metaManager.clear();
```
to clear meta manager for testing recovery

Reply:
Added, thank you